src/server/master_server/lock_manager.cc

```diff
@@ -2,33 +2,57 @@
 
 #include <thread>
 
-#include "absl/hash/hash.h"
-
 using google::protobuf::util::Status;
 using google::protobuf::util::StatusOr;
 
 namespace gfs {
 namespace server {
 
+LockManager::LockManager() {
+  num_of_submap_ = std::max(std::thread ::hardware_concurrency(), 
+                            (unsigned int)1);
+  // TODO(Xi): advanced usage may require tuning the submap number so 
+  // to configure the above number
+  
+  // Initialize the locks and submaps
+  submap_lock_ = std::vector<absl::Mutex*>(num_of_submap_, new absl::Mutex());
+  file_path_locks_ = std::vector<absl::flat_hash_map<
+                         std::string, absl::Mutex*>>(num_of_submap_);
+}
+
+inline size_t LockManager::submap_id(const std::string& filename) const {
```

Comment:
nit: method naming style?

Reply:
Will fix it. Sorry my head is still stuck with starting with a method by a lower case